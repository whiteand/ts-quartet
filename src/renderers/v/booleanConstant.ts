import { ISchemaRenderer } from "../../types";
import { fromValueSchemaAlloc } from "./utils";

export const booleanConstantRenderer: ISchemaRenderer<
  boolean
> = fromValueSchemaAlloc(
  (valueId, schema) => {
    return schema ? `${valueId} === true` : `${valueId} === false`;
  },
  (valueId, schema) => {
    return schema ? `${valueId} !== true` : `${valueId} !== false`;
  }
);
