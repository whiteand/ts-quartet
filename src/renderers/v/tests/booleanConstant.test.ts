import { booleanConstantRenderer } from "../booleanConstant";
import { snapshot } from "./snapshot";

describe("v constant number renderer", () => {
  test("v(true)", () => {
    snapshot(booleanConstantRenderer, { schema: true });
  });
  test("v(false)", () => {
    snapshot(booleanConstantRenderer, { schema: false });
  });
});
