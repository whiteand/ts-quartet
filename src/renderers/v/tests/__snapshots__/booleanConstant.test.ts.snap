// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v constant number renderer v(false) 1`] = `"_valueId === false"`;

exports[`v constant number renderer v(false) 2`] = `"_valueId !== false"`;

exports[`v constant number renderer v(false) 3`] = `"if (_valueId === false) return true;"`;

exports[`v constant number renderer v(false) 4`] = `"if (_valueId !== false) return false;"`;

exports[`v constant number renderer v(true) 1`] = `"_valueId === true"`;

exports[`v constant number renderer v(true) 2`] = `"_valueId !== true"`;

exports[`v constant number renderer v(true) 3`] = `"if (_valueId === true) return true;"`;

exports[`v constant number renderer v(true) 4`] = `"if (_valueId !== true) return false;"`;
